server/src/main/java/org/apache/druid/server/coordinator/helper/DruidCoordinatorSegmentCompactor.java

```diff
@@ -178,12 +178,11 @@ private CoordinatorStats doRun(
       final List<DataSegment> segmentsToCompact = iterator.next();
       final String dataSourceName = segmentsToCompact.get(0).getDataSource();
 
-      if (segmentsToCompact.size() > 1) {
+      if (!segmentsToCompact.isEmpty()) {
```

Comment:
Line#179 should move in this if block.

Reply:
Thank you for finding this! Will fix it.

Reply:
Fixed.